src/socket/socket-validation.ts

```diff
@@ -0,0 +1,11 @@
+import * as jwt from 'jsonwebtoken';
+
+export const socketValidation = async (ctx, next) => {
+    const token = ctx.request.body.id;
```

Comment:
do not store the token in you body .. store it as a cookie ... the schenario should be like this:
the use log to your server via route '/login'
the server put this token in his cookie (read about cookies)
each time the user makes any request he sends this token inside his cookie (it does it automatically if you allow that).
the socket/checkout route gets this token from user's cookie and use/store it to identify this user. 
better if i would explain it to you in person